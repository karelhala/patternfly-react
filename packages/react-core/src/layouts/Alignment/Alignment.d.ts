import { HTMLProps, SFC } from 'react';
import { OneOf } from '../../typeUtils';

export const AlignmentVariant: {
  primary: 'primary';
  secondary: 'secondary';
  tertiary: 'tertiary';
  danger: 'danger';
  link: 'link';
  action: 'action';
};

export interface AlignmentProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
  variant?: OneOf<typeof AlignmentVariant, keyof typeof AlignmentVariant>;
}

declare const Alignment: SFC<AlignmentProps>;

export default Alignment;
