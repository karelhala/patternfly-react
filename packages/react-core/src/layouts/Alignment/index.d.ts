export { default as Alignment, AlignmentProps } from './Alignment';
