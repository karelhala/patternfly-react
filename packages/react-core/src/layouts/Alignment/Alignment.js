import React from 'react';
import styles from '@patternfly/patternfly-next/layouts/Alignment/styles.css';
import PropTypes from 'prop-types';
import { css, getModifier } from '@patternfly/react-styles';

export const AlignmentVariant = {
  left: 'left',
  center: 'center',
  right: 'right'
};

const propTypes = {
  variant: PropTypes.oneOf(Object.keys(AlignmentVariant)),
  children: PropTypes.node,
  className: PropTypes.string
};

const defaultProps = {
  variant: 'left',
  children: null,
  className: null
};

const Alignment = ({ variant, children, className, ...props }) => (
  <div
    {...props}
    className={css(styles.alignment, getModifier(styles, variant), className)}
  >
    {children}
  </div>
);

Alignment.propTypes = propTypes;
Alignment.defaultProps = defaultProps;

export default Alignment;
