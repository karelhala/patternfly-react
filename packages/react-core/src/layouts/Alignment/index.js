export { default as Alignment, AlignmentVariant } from './Alignment';
