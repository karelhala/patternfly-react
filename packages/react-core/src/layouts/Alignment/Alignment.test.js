import React from 'react';
import { shallow } from 'enzyme';
import Alignment, { AlignmentVariant } from './Alignment';

Object.keys(AlignmentVariant).forEach(type => {
  test(`Position ${type}`, () => {
    const view = shallow(<Alignment variant={type}>content</Alignment>);
    expect(view).toMatchSnapshot();
  });
});
