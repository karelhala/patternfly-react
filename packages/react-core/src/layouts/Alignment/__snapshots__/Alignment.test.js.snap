// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Position center 1`] = `
.pf-l-alignment.pf-m-center {
  display: block;
  padding: 0px;
  margin: 0px;
  text-align: center;
}

<div
  className="pf-l-alignment pf-m-center"
>
  content
</div>
`;

exports[`Position left 1`] = `
.pf-l-alignment.pf-m-left {
  display: block;
  padding: 0px;
  margin: 0px;
  text-align: left;
}

<div
  className="pf-l-alignment pf-m-left"
>
  content
</div>
`;

exports[`Position right 1`] = `
.pf-l-alignment.pf-m-right {
  display: block;
  padding: 0px;
  margin: 0px;
  text-align: right;
}

<div
  className="pf-l-alignment pf-m-right"
>
  content
</div>
`;
